patches/kaggle_gcp.py

```diff
@@ -29,26 +29,35 @@ def add_integration(self, integration_name):
     def has_bigquery(self):
         return 'bigquery' in self.integrations.keys()
 
+    def has_gcs(self):
+        return 'gcs' in self.integrations.keys()
+
 
 class KaggleKernelCredentials(credentials.Credentials):
     """Custom Credentials used to authenticate using the Kernel's connected OAuth account.
     Example usage:
     client = bigquery.Client(project='ANOTHER_PROJECT',
                                 credentials=KaggleKernelCredentials())
     """
+    def __init__(self, target=GcpTarget.BIGQUERY):
+        super().__init__()
+        self.target = target
 
     def refresh(self, request):
         try:
             client = UserSecretsClient()
-            self.token, self.expiry = client.get_bigquery_access_token()
+            if self.target == GcpTarget.BIGQUERY:
+                self.token, self.expiry = client.get_bigquery_access_token()
+            elif self.target == GcpTarget.GCS:
+                self.token, self.expiry = client._get_gcs_access_token()
         except ConnectionError as e:
             Log.error(f"Connection error trying to refresh access token: {e}")
             print("There was a connection error trying to fetch the access token. "
-                  "Please ensure internet is on in order to use the BigQuery Integration.")
+                  f"Please ensure internet is on in order to use the {self.target.service} Integration.")
             raise RefreshError('Unable to refresh access token due to connection error.') from e
         except Exception as e:
             Log.error(f"Error trying to refresh access token: {e}")
-            if (not get_integrations().has_bigquery()):
+            if (not get_integrations().has_bigquery() and self.target == GcpTarget.BIGQUERY):
```

Comment:
Maybe instead of making this more specific to BigQuery you could just update the messages below to refer to the target service like you did above and in the condidition do `if (not self.target in get_integrations().keys())`

Reply:
Changed it to use `self.target.service`. However, I couldn't use the same in the condition, because `service` here is human readable string (e.g. "Google Cloud Storage"), which doesn't match to what we have in the `KAGGLE_KERNEL_INTEGRATIONS` variable. 

Lmk if you like the previous version (with the condition) better?

Reply:
Actually, it looks like the condition is needed for the message to make sense. Let me change it back.

Reply:
heheh, we commented at the same time :)

Reply:
Lmk if this one is better? I modified `KaggleIntegrations` to use the newly added enum, and then use it under the error condition.